src/commonMain/kotlin/sim/SimParticipant.kt

```diff
@@ -90,8 +90,12 @@ class SimParticipant(val character: Character, val rotation: Rotation, val sim:
         // Compute initial stats
         recomputeStats()
 
-        // Initialize our subject resource
-        resource = Resource(this)
+        // Initialize our subject resource(s)
+        resource = mutableMapOf()
```

Comment:
We can use a local variable to build the map instead of assigning to the main property right away, and keep the main property non-mutable.

Reply:
Solved.